scripts/modules/MsIndustryLinks/templates/data_source/DataSourceWorkflow.psm1

```diff
@@ -244,6 +245,77 @@ function New-FlowAzureBlobStorageDatasourceWorkflow {
     return $baseTemplate
 }
 
+function New-LogicAppDataSourceWorkflow {
+    param (
+        [Parameter(Mandatory = $true, HelpMessage = "The workflow configuration object.")]
+        [object] $WorkflowConfig
+    )
+
+    $baseTemplate = Get-Content $PSScriptRoot/templates/logicapp_base.json | ConvertFrom-Json
+
+    $dataSourceType = $WorkflowConfig.dataSource.type.ToLower()
+    $isCustomConnector = "customconnector" -eq $dataSourceType
+
+    # Set the workflow parameters for the data source
+    if ($isCustomConnector) {
+        $apiName = $WorkflowConfig.dataSource.properties.name
+    }
+    else {
+        $apiName = Get-ApiName -DataSourceType $dataSourceType
+    }
+    $dataSourceParameters = $WorkflowConfig.dataSource.parameters
+    $dataSourceConnections = @{
+        value = @{
+            $apiName = @{
+                connectionId   = "[resourceId('Microsoft.Web/connections', '$apiName')]"
+                connectionName = $apiName
+                id             = "[subscriptionResourceId('Microsoft.Web/locations/managedApis', location, '$apiName')]"
+            }
+        }
+    }
+    $dataSourceParameters | Add-Member -MemberType NoteProperty -Name '$connections' -Value $dataSourceConnections
+    $baseTemplate.parameters = $dataSourceParameters
+
+    # Set the custom connector properties if data source is CustomConnector
+    $definition = Get-Content $PSScriptRoot/templates/data_source/$dataSourceType/logicapp_$dataSourceType.json | ConvertFrom-Json
+    if ($isCustomConnector) {
```

Comment:
Do you think modularising these into data source types would help make this function easier to read?
There is a lot of custom connector specific configuration that could be split out?

Reply:
Since the custom connector is an exception, I can split out the custom connector code into its own function. I'd rather not repeat code all over the place for each data source type.